Zend/zend_objects.c

```diff
@@ -253,9 +262,18 @@ ZEND_API void ZEND_FASTCALL zend_objects_clone_members(zend_object *new_object,
 		} ZEND_HASH_FOREACH_END();
 	}
 
-	if (old_object->ce->clone) {
+	if (has_clone_method) {
 		GC_ADDREF(new_object);
 		zend_call_known_instance_method_with_0_params(new_object->ce->clone, new_object, NULL);
+
+		for (uint32_t i = 0; i < new_object->ce->default_properties_count; i++) {
+			zend_property_info *prop_info = new_object->ce->properties_info_table[i];
+			if (prop_info && (prop_info->flags & (~ZEND_ACC_STATIC|ZEND_ACC_READONLY)) && ZEND_TYPE_IS_SET(prop_info->type)) {
```

Comment:
`(prop_info->flags & (ZEND_ACC_STATIC|ZEND_ACC_READONLY)) == ZEND_ACC_READONLY`? `~ZEND_ACC_STATIC` would flip most flags on, which is likely not what you want.

Also, I think the `ZEND_TYPE_IS_SET` is unnecessary since types for `readonly` are required at compile-time.